dot/parachain/overseer/overseer.go

```diff
@@ -47,9 +57,10 @@ type BlockState interface {
 	FreeImportedBlockNotifierChannel(ch chan *types.Block)
 	GetFinalisedNotifierChannel() chan *types.FinalisationInfo
 	FreeFinalisedNotifierChannel(ch chan *types.FinalisationInfo)
+	GetRuntime(hash common.Hash) (runtime.Instance, error)
 }
 
-func NewOverseer(blockState BlockState) *Overseer {
+func NewOverseer(blockState BlockState) OverseerSystem {
```

Comment:
```suggestion
func NewOverseer(blockState BlockState) *Overseer {
```

A good practice is to `accept interfaces, return structs`

Reply:
updated.